### PR TITLE
Add option to exclude files based on regular expression

### PR DESCRIPTION
**Pull request by Iambecomeroot**
I wanted to exclude `/node_modules/`.

### CODE REVIEW

**Comment by Iambecomeroot** [comment]
Is this likely to get merged or should I publish a fork?

**Comment by xzyfer** [comment]
Exclusions are likely to be merged. Globbing will not.

**Comment by Iambecomeroot** [comment]
Do you mind if I publish my fork on npm then?

**Comment by xzyfer** [comment]
If you wish :)

**Comment by xzyfer** [comment]
FWIW this implementation isn't complete. It's important to also do the exclusion when determine the ancestors and descendants. You'd also probably want to cache the regex check in the constructor to avoid doing it many times.

**Comment by Iambecomeroot** [comment]
I didn't want to check ancestors and descendants because I assumed the usual use case would be `/node_modules/` and I wanted to keep things simple. To add the check to ancestors and descendants would it just be as simple as doing the same check in `Graph.prototype.visitAncestors` and `Graph.prototype.visitDescendents`?

What do you mean by caching the regex check? You mean `this.exclude instanceof RegExp`?

**Comment by Iambecomeroot** [comment]
Something like that?

**Comment by xzyfer** [comment]
Not dealing with ancestors and descendants will mean that if a file imports a file from `node_modules` is appear in the dependency graph. This may be your intention but it's ["surprising"][1] and breaks the primary usecase of this library - the node-sass watcher.

[1]: https://en.wikipedia.org/wiki/Principle_of_least_astonishment

**Comment by xzyfer** [comment]
If files and/or directories are being excluded there should be no way for those exclude to appear in the graph.

**Comment by Iambecomeroot** [comment]
That makes sense. Is what I suggested the best way to do that?

**Comment by xzyfer** [comment]
The regex caching is better. You'll still need to prevent exclude files being referenced in as [ancestors][1] and [descendants][2].

[1]: https://github.com/xzyfer/sass-graph/blob/master/sass-graph.js#L95
[2]: https://github.com/xzyfer/sass-graph/blob/master/sass-graph.js#L80

**Comment by Iambecomeroot** [comment]
I'll write some tests I guess.

**Comment by xzyfer** [comment]
Please remove package-lock.json

**Comment by Iambecomeroot** [comment]
Sure thing. BTW you should have write access, no?

**Comment by xzyfer** [comment]
I do but I'm mostly catching up whilst in bed. Open source isn't my day job :)

**Comment by xzyfer** [comment]
This is looking good, thanks for your time. 

I have some performance concerns, but they can be address separately. I'll get this released in the next day or so.

File: readme.md [hunk 1]
```diff
@@ -82,6 +82,13 @@ Default: `false`
 
 Follow symbolic links.
 
+#### excludes
```
Comment (xzyfer) [inline readme.md]: The option is exclude, not excludes

Comment (Iambecomeroot) [inline readme.md]: Oops

**Comment by Iambecomeroot** [comment]
Yeah it's true that it checks a lot of regexes but idk how to avoid that.

**Comment by Iambecomeroot** [comment]
Any update on this?

**Comment by Iambecomeroot** [comment]
What are we waiting on? I'd like to make my fork after this has been merged.

**Comment by xzyfer** [comment]
Thanks @Iambecomeroot. Apologies for the delay I've be occupied with other projects.

**Pull request merged**